src/App.jsx

```diff
@@ -14,70 +16,73 @@ export const goods = [
   'Garlic',
 ];
 
-export const App = () => (
-  <main className="section container">
-    <h1 className="title is-flex is-align-items-center">No goods selected</h1>
+const DEFAULT_GOOD = 'Jam';
+const CLEARED_SELECTED_GOOD = '';
+const SELECT_BUTTON_VALUE = '+';
+const CLEAR_BUTTON_VALUE = '-';
 
-    <h1 className="title is-flex is-align-items-center">
-      Jam is selected
+export const App = () => {
+  const [selectedGood, setSelectedGood] = useState(DEFAULT_GOOD);
 
-      <button
-        data-cy="ClearButton"
-        type="button"
-        className="delete ml-3"
-      />
-    </h1>
+  return (
+    <main className="section container">
+      <h1 className="title is-flex is-align-items-center">
+        {selectedGood
+          ? `${selectedGood} is selected`
+          : `No goods selected`
+        }
 
-    <table className="table">
-      <tbody>
-        <tr data-cy="Good">
-          <td>
-            <button
-              data-cy="AddButton"
-              type="button"
-              className="button"
-            >
-              +
-            </button>
-          </td>
+        {selectedGood && (
+          <button
+            data-cy="ClearButton"
+            type="button"
+            className="delete ml-3"
+            onClick={() => setSelectedGood(CLEARED_SELECTED_GOOD)}
+          />
+        )}
+      </h1>
 
-          <td data-cy="GoodTitle" className="is-vcentered">
-            Dumplings
-          </td>
-        </tr>
+      <table className="table">
+        <tbody>
+          {goods.map((good) => {
+            const isSelectedGood = selectedGood === good;
 
-        <tr data-cy="Good" className="has-background-success-light">
-          <td>
-            <button
-              data-cy="RemoveButton"
-              type="button"
-              className="button is-info"
-            >
-              -
-            </button>
-          </td>
+            return (
+              <tr
+                key={good}
+                data-cy="Good"
+                className={
+                  cn({ 'has-background-success-light': isSelectedGood })
+                }
+              >
+                <td>
+                  <button
+                    data-cy="AddButton"
```

Comment:
here you need add to data-cy atribute 'Remove button'. that why you have this problem with test 'should show RemoveButton for a selected good'.

Reply:
thnks bro, you made my day)) every time I understand that every trouble or bug doesn't come from nowhere, and no metter what happend, I'm doing the same mistakes 💪